### PR TITLE
Expose network list endpoint

### PR DESCRIPTION
**Pull request by shahnami**
Add a `listNetworks` function to the sentinel package

I've added the endpoint to sentinel package to stay consistent with where it is hosted in `platform-sdk` (in monitor).

### CODE REVIEW

File: packages/sentinel/README.md [hunk 1]
```diff
@@ -373,6 +373,16 @@ await client.pause('8181d9e0-88ce-4db0-802a-2b56e2e6a7b1');
 await client.unpause('8181d9e0-88ce-4db0-802a-2b56e2e6a7b1');
 ```
 
+### List Networks
+
+To list tenant enabled networks, you can call the `listNetworks` function on the client, which returns a `Network[]` object:
+
+```js
+await client.listNetworks(); // lists all networks
+await client.listNetworks('prod'); // lists only production networks
```
Comment (zeljkoX) [inline packages/sentinel/README.md:382]: late to the game, but in crypto world do users use terms mainnet and testnet instead of prod and test?

**Review by zeljkoX — COMMENTED** [review]
LGTM

File: packages/sentinel/README.md (hunk 1 above)
Comment (shahnami) [reply packages/sentinel/README.md:382]: Want to be consistent with our UI where we also refer to production and test networks. Also mainnet could be confused with _just_ homestead, whereas we would be referring to all mainnet networks.

**Review by BjMrq — APPROVED** [review]
🦘

File: packages/sentinel/src/models/networks.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export type NetworkType = 'prod' | 'test';
```
Comment (BjMrq) [inline packages/sentinel/src/models/networks.ts:1]: nits eventualy would use full words like "production" if not "mainnet" mentioned by Ze

**Comment by mok0230** [comment]
Same as https://github.com/OpenZeppelin/platform-sdk/pull/17#discussion_r1271547898

**Review by mok0230 — APPROVED** [review]
lgtm

**Pull request merged**